app/assets/javascripts/actions/campaign_actions.js

```diff
@@ -91,12 +93,13 @@ export const addCampaign = (courseId, campaignId) => dispatch => {
   );
 };
 
-const fetchAllCampaignsPromise = () => {
+const fetchAllCampaignsPromise = (courseCampaigns) => {
   return new Promise((res, rej) => {
     return $.ajax({
       type: 'GET',
       url: `/lookups/campaign.json`,
       success(data) {
+        data.values = _.difference(data.values, courseCampaigns);
```

Comment:
something like this? @ragesoss

Reply:
I was thinking that the action would just pass the values to the reducer, without worrying about the interaction between course campaigns. Then a selector (ie, a reselect function like with the student list) would be used in mapStateToProps to send the unused campaigns to the component that displays the options for adding a new one.